python/plugins/processing/core/parameters.py

```diff
@@ -1186,15 +1190,17 @@ def getAsScriptCode(self):
         param_type = ''
         if self.optional:
             param_type += 'optional '
-        param_type += 'string'
-        return '##' + self.name + '=' + param_type + self.default
+        param_type += 'string '
+        return '##' + self.name + '=' + param_type + str(self.default)
 
     @classmethod
     def fromScriptCode(self, line):
         isOptional, name, definition = _splitParameterOptions(line)
         descName = _createDescriptiveName(name)
         if definition.lower().strip().startswith('string'):
-            default = definition.strip()[len('string') + 1:]
+            default = definition.strip()[len('string') + 1:] or None
+            if default == 'None':
+                default = None
```

Comment:
If I'm not wrong, there were discussion about using `None` as default value. IMHO there should be a difference between a parameter with an empty value and `None` value. Maybe even, when the parameter is `None`, there should be no default value and the algorithm should not use it at all.